Objects/longobject.c

```diff
@@ -6685,3 +6682,78 @@ Py_ssize_t
 PyUnstable_Long_CompactValue(const PyLongObject* op) {
     return _PyLong_CompactValue((PyLongObject*)op);
 }
+
+const PyLongLayout PyLong_LAYOUT = {
+    .bits_per_digit = PyLong_SHIFT,
+    .word_endian = PY_LITTLE_ENDIAN ? -1 : 1,
+    .array_endian = -1,  // least significant first
+    .digit_size = sizeof(digit),
+};
+
+
+int
+PyLong_Export(PyObject *obj, PyLong_DigitArray *array)
+{
+    if (!PyLong_Check(obj)) {
+        PyErr_Format(PyExc_TypeError, "expect int, got %T", obj);
+        return -1;
+    }
+    PyLongObject *self = (PyLongObject*)obj;
+
+    array->obj = Py_NewRef(obj);
+    array->negative = _PyLong_IsNegative(self);
+    array->ndigits = _PyLong_DigitCount(self);
+    if (array->ndigits == 0) {
+        array->ndigits = 1;
+    }
+    array->digits = self->long_value.ob_digit;
+    return 0;
+}
+
+
+void
+PyLong_ReleaseExport(PyLong_DigitArray *array)
+{
+    Py_CLEAR(array->obj);
+    array->negative = 0;
+    array->ndigits = 0;
+    array->digits = NULL;
+}
+
+
+/* --- PyLongWriter API --------------------------------------------------- */
+
+PyLongWriter* PyLongWriter_Create(int negative, size_t ndigits, Py_digit **digits)
+{
+    if (ndigits > (size_t)PY_SSIZE_T_MAX) {
+        PyErr_NoMemory();
+        return NULL;
+    }
```

Comment:
``_PyLong_New()`` has similar check, but with a lower value ``MAX_LONG_DIGITS``.  Probably this is redundant.

Reply:
_PyLong_New() parameter is signed, whereas PyLongWriter_Create() parameter is unsigned. That's why a test is needed, to avoid integer overflow in the implicit cast. I just made the cast to Py_ssize_t explicit.